### PR TITLE
Incorporating design review feedback from Derek

### LINKED ISSUE #165: Feedback from Derek ahead of our demo
### General

- [x] Create ACS geographies on staging site by running `python manage.py import_data`
- [x] Run `python manage.py initialize_pldp` on staging site
- [x] Bump the size of header-level help text
- [x] Rename “deactivate” to “delete,” with help text explaining that it’s a soft delete

### List views
- [x] Add explainer text for top-level entities (Studies, Surveys, Study Areas, Locations)
- [x] Add “Create new” button to all list view templates
- [ ] Combine list table buttons into “Actions” column on list views (moved to #168)

### Navbar
- [x] Change “Create new…” to “Create survey”
- [x] Move DataMade site credit to the footer

### Surveys
- [x] Figure out if Study and Location should be required fields; update the data handler appropriately (Update: they should be required!)

### Adding survey questions view
- [x] Display the question options when adding a preset question to a survey
- [x] Change “Label” -> “Question text”
- [x] Change “Default” -> “Default answer”
- [x] Add help text for each form element

### Survey edit view

- [ ] Add dragging icon to draggable survey questions. (Still to do: Make sure that the draggable cursor appears on the left side of the question form on Survey Edit; moved to #169)
- [x] Change “Save ordering” button to “Save and exit”
- [x] Add a “Preview” button
- [x] Make Survey.type not editable on the properties page

### Run Survey view

- [x] Make Run Survey icon into a button
- [x] Add to table metadata:
    - [x] Number of questions
    - [x] Last run
- [x] Add the name of the survey to “Survey submitted!” confirmation page

### Survey submitted list view
- [x] Show number of runs of the survey

### Survey submitted detail view
- [x] Add help text to Census geographies selectbox explaining that you can select multiple
- [x] Clean up styles of Analyze form (Select2, bootstrap inputs)
- [x] Change “Primary source” label to say “Survey question”

### PR DESCRIPTION
**Pull request by beamalsky**
## Overview

Connects #165. This PR closes the various fixes requested by Derek and enumerated in #165.

## Testing

@jeancochrane this is a lot of little changes and I don't know if everything needs a second pair of eyes. But here are the things accomplished in this PR that could be checked: 

### General

- [x] Create ACS geographies on staging site by running `python manage.py import_data` (Census geographies should show up on https://justspaces.datamade.us/)
- [x] Run `python manage.py initialize_pldp` on staging site (Languages and countries should show up on https://justspaces.datamade.us/)
- [x] Bump the size of header-level help text
- [x] Rename “deactivate” to “delete,” with help text explaining that it’s a soft delete

### List views
- [x] Add explainer text for top-level entities (Studies, Surveys, Study Areas, Locations)
- [x] Add “Create new” button to all list view templates
- [x] Combine list table buttons into “Actions” column on list views

### Navbar
- [x] Change “Create new…” to “Create survey”
- [x] Move DataMade site credit to the footer

### Surveys
- [x] Make Study and Location required fields

### Adding survey questions view
- [x] Display the question options when adding a preset question to a survey
- [x] Change “Label” -> “Question text”
- [x] Change “Default” -> “Default answer”
- [x] Add help text for each form element

### Survey edit view

- [x] Add dragging icon to draggable survey questions
- [x] Change “Save ordering” button to “Save and exit”
- [x] Add a “Preview” button
- [x] Make Survey.type not editable on the properties page

### Run Survey view

- [x] Make Run Survey icon into a button
- [x] Add to table metadata:
    - [x] Number of questions
    - [x] Last run
- [x] Add the name of the survey to “Survey submitted!” confirmation page

### Survey submitted list view
- [x] Show number of runs of the survey

### Survey submitted detail view
- [x] Add help text to Census geographies selectbox explaining that you can select multiple
- [x] Clean up styles of Analyze form (Select2, bootstrap inputs)
- [x] Change “Primary source” label to say “Survey question”

### CODE REVIEW

**Comment by beamalsky** [comment]
Ok @jeancochrane this is ready for your review! It completes all the tasks in #165 except for ~1.5, which I've broken off into separate issues and notes in that original issue. See the checklist above for what this accomplishes.

File: surveys/templates/partials/sidebar.html [hunk 1]
```diff
@@ -71,7 +74,11 @@ <h3><b>Just Spaces</b></h3>
             <div><a class="btn-sidebar" href="{% url 'signup' %}">Create new account</a></div>
             {% endif %}
             <br />
-            <div class="about-blurb"><strong>Just Spaces</strong> is a tool from <a class="about-link" href="https://www.universitycity.org/" target=_blank>University City District</a> and <a class="about-link" href="https://datamade.us/" target=_blank>DataMade</a> to promote better and more just public spaces</div>
+            <div class="about-blurb">
+              <strong>Just Spaces</strong> is a tool from <a class="about-link" href="https://www.universitycity.org/" target=_blank>University City District</a> to promote better and more just public spaces
+            </div>
+            <br />
+            <div>Built by <a class="about-link" href="https://datamade.us/" target=_blank>DataMade</a></div>
```
Comment (jeancochrane) [inline surveys/templates/partials/sidebar.html]: Our typical footer text is `Website by DataMade [and partner]`, see e.g. https://largelots.org/.

Comment (jeancochrane) [reply surveys/templates/partials/sidebar.html]: Derek signed off on `Built by`, so I'm resolving this.

File: surveys/templates/survey_list_edit.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  <table id="survey-list-edit" class="table table-hover table-striped table-responsive-sm card-text">
+  <thead>
+    <tr>
+      <th scope="col">Survey Name</th>
+      <th scope="col">Last Updated</th>
+      <th scope="col">Type</th>
+      <th class="text-center" scope="col">Edit</th>
+      <th class="text-center" scope="col">Preview</th>
+      <th class="text-center" scope="col">Publish</th>
+      <th class="text-center" scope="col">Delete</th>
```
Comment (jeancochrane) [inline surveys/templates/survey_list_edit.html]: I like the new styles for these icons a lot! The columns are still labelled with their specific actions, however, instead of being grouped under one heading for `Actions`:

Screen Shot 2019-06-11 at 5 23 26 PM

Was this what you were intending? Maybe we could have a quick convo IRL to talk about ways of condensing these actions.

File: fobi_custom/override_theme/templates/override_theme/form_entry_submitted_ajax.html [hunk 3]
```diff
@@ -1,9 +1,10 @@
 <div class="col col-lg-6 card">
   <div class="card-body">
-    <h2 class="card-title">Survey submitted!</h2>
+    <h2 class="card-title">{{form_entry}} submitted!</h2>
```
Comment (jeancochrane) [inline fobi_custom/override_theme/templates/override_theme/form_entry_submitted_ajax.html]: I like that we're displaying the title of the survey now, but I found it a bit hard to parse with a long survey name:

Screen Shot 2019-06-11 at 5 24 57 PM

What about something like:

> Survey _"Test saving ACS data"_ submitted!

File: surveys/templates/partials/sidebar.html [hunk 4]
```diff
@@ -46,6 +32,23 @@ <h3><b>Just Spaces</b></h3>
                     View collected data
                 </a>
             </li>
+            <li>
+                <a href="#createSurveySubmenu" data-toggle="collapse" aria-expanded="false" class="dropdown-toggle">
+                    <i class="fas fa-sm fa-home"></i>
```
Comment (jeancochrane) [inline surveys/templates/partials/sidebar.html]: What do you think about `fa-gear` here? `fa-home` doesn't feel quite right, but it's hard to think of a good alternative for `Admin`.

File: surveys/templates/partials/sidebar.html [hunk 5]
```diff
@@ -46,6 +32,23 @@ <h3><b>Just Spaces</b></h3>
...
+                        <a href="{% url 'agencies-create' %}">- Create Agency</a>
+                    </li>
+                    <li>
+                        <a href="{% url 'studies-create' %}">- Create Study</a>
+                    </li>
+                    <li>
+                        <a href="{% url 'locations-create' %}">- Create Location</a>
+                    </li>
+                </ul>
+            </li>
```
Comment (jeancochrane) [inline surveys/templates/partials/sidebar.html]: I like this new menu a lot. Since our list views all have `Create new` buttons now, what do you think about having these links point to the list pages instead (such that the menu options would be `Agencies | Studies | Locations`)? My instinct is that the list view is more generally useful than the create view.

File: surveys/templates/survey_list_edit.html (hunk 2 above)
Comment (beamalsky) [reply surveys/templates/survey_list_edit.html]: I opened a new issue for that here: #168 

I started to condense them into one column and couldn't find a way that didn't feel pretty cluttered and confusing. Do you know of any example sites I could reference?

File: fobi_custom/override_theme/templates/override_theme/form_entry_submitted_ajax.html (hunk 3 above)
Comment (beamalsky) [reply fobi_custom/override_theme/templates/override_theme/form_entry_submitted_ajax.html]: Good idea!

File: surveys/templates/partials/sidebar.html (hunk 4 above)
Comment (beamalsky) [reply surveys/templates/partials/sidebar.html]: Love it

Comment (beamalsky) [reply surveys/templates/partials/sidebar.html]: Update: It's `fa-cog`

File: surveys/templates/survey_list_edit.html (hunk 2 above)
Comment (jeancochrane) [reply surveys/templates/survey_list_edit.html]: Cool! I'm fine pushing off to #168 since you tried a few options and didn't like them as much.

File: surveys/templates/partials/sidebar.html (hunk 5 above)
Comment (beamalsky) [reply surveys/templates/partials/sidebar.html]: I think that's a great idea. I'd been meaning to create basic List, Detail, and Delete views for Agency too, and I did that here so that this will work.

File: surveys/views.py [hunk 6]
```diff
@@ -31,6 +31,50 @@ class AgencyCreate(CreateView):
...
+    def get_context_data(self, **kwargs):
+        context = super(AgencyDetail, self).get_context_data(**kwargs)
+
+        agency = context['agency']
+
+        context['rows'] = [
+            ('Department', agency.department),
+            ('Phone', agency.phone),
+            ('Type', agency.type),
+            ('Language', agency.language.name_en),
```
Comment (jeancochrane) [inline surveys/views.py]: Are Agencies required to have a `language` attribute? I have an agency in my database that doesn't have a `language` so this lookup returns an error.

Comment (beamalsky) [reply surveys/views.py]: Oh thanks for the catch! Yes, language is optional. Now fixed.

File: surveys/views.py [hunk 7]
```diff
@@ -28,7 +28,7 @@ class AgencyCreate(CreateView):
     form_class = survey_forms.AgencyCreateForm
     model = pldp_models.Agency
     template_name = "agency_create.html"
-    success_url = '/'
+    success_url = 'agencies-list'
```
Comment (jeancochrane) [inline surveys/views.py]: Does this need to be wrapped in a `reverse_lazy` call? I got redirected to `agencies/create/agencies-list`, which returns 404.

Comment (beamalsky) [reply surveys/views.py]: Ee sorry I should have tested that! You're right.

File: fobi_custom/plugins/form_elements/fields/utils.py [hunk 8]
```diff
@@ -4,3 +4,13 @@ def choices_to_help_text(choices):
         help_text += '- ' + choice + '<br />'
 
     return help_text
+
+
+def choices_to_help_text_skip_first(choices):
```
Comment (jeancochrane) [inline fobi_custom/plugins/form_elements/fields/utils.py]: Another way to implement this could have been to add a `skip_first` keyword argument to the parameter list of `choices_to_help_text` and set it to default to `False`, e.g.:

```python
def choices_to_help_text(choices, skip_first=False):
    ...
```

That way all of the code that formats choices for display would share the same function, and if you ever wanted to change the input or output format you'd only have to adjust one function.

File: surveys/static/js/survey_analysis.js [hunk 9]
```diff
@@ -0,0 +1,209 @@
+$(document).ready(function () {
```
Comment (jeancochrane) [inline surveys/static/js/survey_analysis.js]: I like the idea of transferring this logic to its own file, but it looks like this code is duplicated in the `survey_submitted_detail.html` template. Did you mean to keep them both?

File: surveys/views.py [hunk 10]
```diff
@@ -342,14 +387,24 @@ def get_context_data(self, **kwargs):
...
     context_object_name = 'surveys'
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context['surveys'] = context['surveys'].exclude(active=False).filter(published=True).order_by('-updated')
         context['published'] = True
 
+        for survey in context['surveys']:
+            survey_questions = fobi_models.FormElementEntry.objects.all().filter(form_entry_id=survey.id)
+            survey.question_count = len(survey_questions)
```
Comment (jeancochrane) [inline surveys/views.py]: Let's use [`count()`](https://docs.djangoproject.com/en/2.1/ref/models/querysets/#django.db.models.query.QuerySet.count) here instead of `len()`. From [the docs](https://docs.djangoproject.com/en/2.1/ref/models/querysets/):

> A QuerySet is evaluated when you call len() on it. This, as you might expect, returns the length of the result list.
>
> Note: If you only need to determine the number of records in the set (and don’t need the actual objects), it’s much more efficient to handle a count at the database level using SQL’s SELECT COUNT(*). Django provides a count() method for precisely this reason.

File: surveys/views.py [hunk 11]
```diff
@@ -369,6 +424,9 @@ def get_context_data(self, **kwargs):
             except SurveyFormEntry.DoesNotExist:
                 survey.form_title = "[Deleted Survey]"
 
+            survey_submissions = surveys.filter(form_id=survey.form_id)
+            survey.times_run = len(survey_submissions)
```
Comment (jeancochrane) [inline surveys/views.py]: Another good candidate for `count()`.

File: fobi_custom/plugins/form_elements/fields/utils.py (hunk 8 above)
Comment (beamalsky) [reply fobi_custom/plugins/form_elements/fields/utils.py]: Much more elegant 😎
